### PR TITLE
click to show compiled vega spec in vega-lite mode

### PR DESCRIPTION
**Pull request by starry97**

### CODE REVIEW

**Comment by domoritz** [comment]
Screenshot? 😄

**Comment by arvind** [comment]
Or even better: GIF with [LICEcap](http://www.cockos.com/licecap/)!

**Comment by starry97** [comment]
screen shot 2017-01-24 at 1 25 10 pm
screen shot 2017-01-24 at 1 24 57 pm
@domoritz

**Comment by starry97** [comment]
untitled

**Comment by mathisonian** [comment]
Thanks @starry97, this is looking good!

A few things:
* Can we update it so that the compiled spec is resizable (taking advantage of the [react-split-pane](https://github.com/tomkp/react-split-pane) infrastructure that is already in the project?
* Can we add a button to the bar when the compiled spec display is open, letting users edit the compiled vega spec? this would involve dispatching an `UPDATE_VEGA_SPEC` action to set the new specification. 
* I'll add a few code style comments inline 

Here are some designs to get the bar styles matching the rest of the editor a little bit more 

closed
open

File: src/components/input-panel/compiled-spec-display/index.js [hunk 1]
```diff
@@ -1,15 +1,28 @@
...
+
+  const toggleStyle = {
+    position: 'absolute',
+    bottom: '0px',
+    color: 'white',
+    cursor: 'pointer',
+    display: 'inline-block',
+    fontSize: '15px',
+    width: '100%',
+    height: '20px',
```
Comment (mathisonian) [inline src/components/input-panel/compiled-spec-display/index.js]: When defining pixel values on a react style object, just use a number and omit `px`, so this becomes: `height: 20`

File: src/components/input-panel/compiled-spec-display/index.js [hunk 2]
```diff
@@ -19,30 +32,52 @@ class CompiledSpecDisplay extends React.Component {
...
+            onClick={this.handleClick.bind(this)}>Read only, click to hide spec </div>
+            <MonacoEditor 
+              mode='json'
+              theme='github'
+              showGutter={true}
+              key={JSON.stringify(this.state)}
+              width={'100%'}
+              height={this.state.height}      
+              // value={JSON.stringify(this.props.value)}
+              value={this.props.value}
```
Comment (mathisonian) [inline src/components/input-panel/compiled-spec-display/index.js]: The properties for `MonacoEditor` are slightly different from the properties that the old `AceEditor` expected. Check out https://github.com/vega/vega-editor/blob/redux/src/components/input-panel/spec-editor/renderer.js#L61-L68 to make sure that syntax highlighting, etc. works

File: src/components/input-panel/index.js [hunk 3]
```diff
@@ -27,9 +27,13 @@ class InputPanel extends React.Component {
...
-      outerComponent = React.createElement(SplitPane, {
-        split: 'horizontal'
-      }, innerPanes);
+      if (this.props.mode === MODES.VegaLite) {
+        outerComponent = React.createElement('div', {style: {width: '100%'}}, innerPanes);
+      } else {
+        outerComponent = React.createElement(SplitPane, {
+          split: 'horizontal'
+        }, innerPanes);
+      }
```
Comment (mathisonian) [inline src/components/input-panel/index.js]: Would be awesome if we could get this working with the `SplitPane` and not have to special case when mode is VegaLite

**Review by mathisonian — COMMENTED** [review]
I just added a couple notes inline

**Comment by starry97** [comment]
untitled
updated version

**Comment by mathisonian** [comment]
@starry97 there are a couple types on the imports for `compiled-spec-header` (it is written as `complied` in the code)

also, I am getting errors when trying to import that file, I think maybe it wasn't checked in

**Comment by starry97** [comment]
@mathisonian My bad. I've pushed the file and fixed the typos.

**Comment by mathisonian** [comment]
cool, thanks!

**Pull request merged**